### PR TITLE
Feature/middleware

### PR DESCRIPTION
**Pull request by antonio-1799**
- Rework middleware for access and refresh token
- Add test endpoint (GET /me) that uses middleware for access token
- Add helper (Enums, Seeders)
- Add Middleware (Validates token and user from the validated token) and Token Service (gets user id from the request context)

### CODE REVIEW

File: src/functions/apis/apis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # MIDDLEWARES
-# - ${file(./src/functions/middlewares/authorization/config.yml)} # Uncomment this if you want to implement middleware
+ - ${file(./src/functions/middlewares/access_token_authorization/config.yml)} # Uncomment this if you want to implement middleware
```
Comment (cjchua1989) [inline src/functions/apis/apis.yml:2]: Please also include the refresh_token authorizer

File: src/functions/apis/auth/action.ts [hunk 2]
```diff
@@ -2,8 +2,17 @@ import { Connection } from 'typeorm';
 import { UserRepository } from '../../../repositories/UserRepository';
 import { AuthAccessDenied } from './responses';
 import * as bcrypt from 'bcrypt';
+import { TokenData } from '../../../services/MiddlewareService';
+import { JWT } from '../../../libs/JWT';
+import { TokenType } from '../../../helper/Enums';
 import { UserModel } from '../../../models/UserModel';
 
+interface LoginResult {
```
Comment (cjchua1989) [inline src/functions/apis/auth/action.ts:10]: Please rename this to AuthResponse

File: src/functions/apis/auth/action.ts [hunk 3]
```diff
@@ -13,11 +22,30 @@ export class AuthAction {
...
     }
 
-    async execute(username: string, password: string): Promise<UserModel> {
+    async execute(username: string, password: string): Promise<LoginResult> {
         const user = await this.repository.getUserByUserName(username);
         if (!user) throw new AuthAccessDenied();
 
         if (!bcrypt.compareSync(password, user.password)) throw new AuthAccessDenied();
-        return user;
+
```
Comment (cjchua1989) [inline src/functions/apis/auth/action.ts:30]: Please make the generation of the access token and refresh token part of the TokenService

File: src/functions/apis/apis.yml (hunk 1 above)
Comment (antonio-1799) [reply src/functions/apis/apis.yml]: resolved

File: src/functions/apis/auth/action.ts (hunk 2 above)
Comment (antonio-1799) [reply src/functions/apis/auth/action.ts]: resolved

File: src/functions/apis/auth/action.ts (hunk 3 above)
Comment (antonio-1799) [reply src/functions/apis/auth/action.ts:32]: resolved, please check new commit thanks

File: src/functions/apis/auth/handler.ts [hunk 4]
```diff
@@ -13,15 +13,17 @@ export async function execute(event: ApiGatewayEvent): Promise<APIHttpResponse>
         const request: AuthRequest = Validate(JSON.parse(event.body));
         const connection = await Databases.getConnection();
         const action = new AuthAction(connection);
-        const user = await action.execute(request.username, request.password);
+        const result = await action.execute(request.username, request.password);
```
Comment (cjchua1989) [inline src/functions/apis/auth/handler.ts:16]: Please change this to 
`const {.user, acceess_token, refresh_token } = await action.execute(request.username, request.password);`

File: src/functions/apis/auth/handler.ts [hunk 5]
```diff
@@ -13,15 +13,17 @@ export async function execute(event: ApiGatewayEvent): Promise<APIHttpResponse>
...
 
         return API_RESPONSE({
             ...Responses.STATUS_200,
             user: {
```
Comment (cjchua1989) [inline src/functions/apis/auth/handler.ts:20]: Apply changes here

**Pull request merged**